safenode/src/client/wallet.rs

```diff
@@ -73,14 +75,121 @@ impl SendClient for Client {
                 fee_ciphers,
             };
 
-            let _responses = self
+            let responses = self
                 .send_to_closest(Request::Cmd(cmd))
                 .await
                 .map_err(|err| Error::CouldNotSendTokens(err.to_string()))?;
 
-            // TODO: validate responses
+            // Get all Ok results of the expected response type `Spend`.
+            let ok_responses: Vec<_> = responses
+                .iter()
+                .flatten()
+                .flat_map(|resp| {
+                    if let Response::Cmd(CmdResponse::Spend(Ok(()))) = resp {
+                        Some(())
+                    } else {
+                        println!("Spend error {resp:?}.");
+                        None
+                    }
+                })
+                .collect();
+
+            // We require a majority of the close group to respond with Ok.
+            if ok_responses.len() >= close_group_majority() {
+                continue;
+            } else {
+                return Err(Error::CouldNotVerifyTransfer(format!(
+                    "Not enough close group nodes accepted the spend. Got {}, required: {}.",
+                    ok_responses.len(),
+                    close_group_majority()
+                )));
+            }
         }
 
         Ok(transfer)
     }
 }
+
+#[async_trait::async_trait]
+impl VerifyingClient for Client {
+    async fn verify(&self, dbc: &Dbc) -> Result<()> {
+        let mut received_spends = std::collections::BTreeSet::new();
+
+        for spend in &dbc.signed_spends {
+            let address = dbc_address(spend.dbc_id());
```

Comment:
This fn seems quite big. Can we split apart the logic into some smaller fns for readability please?

Perhaps just a `verify_spend` function and `verify_response`?

Reply:
I'm moving the VerifyingClient to it's own file, which should help with focus when reading that.

Reply:
And clarified what we are actually getting from the network.

Reply:
fixed